types/test.ts

```diff
@@ -225,20 +228,20 @@ function testArrayArgument() {
   const selector = createSelector([
     (state: {foo: string}) => state.foo,
     (state: {foo: string}) => state.foo,
-    (state: never, props: {bar: number}) => props.bar,
+    (state, props: {bar: number}) => props.bar,
```

Comment:
Implicit `any` values can be used in place of `never` for heterogenous `State` selectors.